rmg: init at 0.5.4

slam-bert opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Adds [Rosalie's Mupen GUI](https://github.com/Rosalie241/RMG) to nixpkgs

NOTE: This project ships it's own mupen64plus-core instead of using the one that's already been packaged. It also comes
with plugins (such as GLideN64) that are compiled during the build phase.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

I've tested the program by changing plugins and playing through small chunks of some games.

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

bachp reviewed (changes requested):
@slam-bert Thanks for the contribution. Two remarks from my side:

Could you sqash your two commits into one and also use `
rmg: init at 0.5.4` as the commit message.

Also just adding a `default.nix` for the package to where you added it doesn't make it available in nixpkgs. You either have to make an entry in `all-packages.nix`. Or even better move it to the [`by-name`](https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name) hirarchy. This is quite new and there is is enough to just add it as you did. Check the [README](https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/README.md) for where exactly to put it.

slam-bert committed:
rmg: init at 0.5.4

rmg: append to init at 0.5.4

rmg: further add to init at 0.5.4

rmg: (hopefully) finally finish init at 0.5.4

rmg: fix qt imports for init at 0.5.4

rmg: move 'which' to nativeBuildInputs for init at 0.5.4

rmg: try only importing 'which' in init at 0.5.4

rmg: fix a whitespace error in init at 0.5.4

rmg: remove 'dash' for init at 0.5.4

rmg: fix up qt packages for init at 0.5.4

rmg: remove seperate 'wrapQtAppsHook' from imports for init at 0.5.4

rmg: add myself to 'maintainer-list.nix' for init at 0.5.4

slam-bert commented:
Hello!

I wasn't able to squash the commits (much to my frustration), so I instead decided to rollback my fork to before the commits and create a new commit. I made sure to move the `default.nix` file to `by-name`, as you recommended.

If any other changes are needed, please let me know!

bachp commented:
@slam-bert I forgot that in the by-name tree the file needs to be named `package.nix` instead of `default.nix`. That's why the CI is failing.

slam-bert commented:
@bachp Thanks for bringing that to my attention! I renamed `default.nix` to `package.nix` and cleaned up the code a bit.

bachp commented on pkgs/by-name/rm/rmg/package.nix:
```diff
@@ -0,0 +1,86 @@
+    speexdsp
+    vulkan-headers
+    vulkan-loader
+    which
```
This doesn't exsit. `which` is usually a shell built in.

slam-bert replied:
`mupen64plus-core` seems to have a dependency on GNU `which`. When I was testing the package using `nix-build` on my system, I couldn't build `mupen64plus-core` without `which` as a buildInput. I've moved it over to nativeBuildInput, which I think is more suitable.

bachp replied:
@slam-bert Did you build this locally on your machine? For example with `nix-build -A rmg` from inside your nixpkgs directory.

slam-bert replied:
@bachp That's what I did. I was able to declare `which` as a dependency and a `nativeBuildInput` just fine.

bachp replied:
Did you check everything in corretly, as you can see from CI this still doesn't evaluate properly.

slam-bert replied:
I'm not sure if I did or not. I saw that the `mupen64plus` nixpkgs already has uses `which` as a `buildInput`, so the errors I've been reading from `ofborg` are perplexing to me.

slam-bert commented:
@ofborg build rmg

slam-bert commented:
`ofborg`has `rmg.passthru.tests on aarch64-darwin` and `rmg on aarch64-darwin`running for some reason, even though I marked the `platforms` in `meta` as `platforms.linux`. When I visited the `ofborg` web interface, I found that the two checks seem to be stuck on `Subscribing to "nixos/nixpkgs.279958"...`. Is there a way to fix this?
